src/main/java/org/spongepowered/common/config/SpongeConfig.java

```diff
@@ -127,30 +181,130 @@ public void reload() {
         }
 
         try {
-            this.root = this.loader.load(ConfigurationOptions.defaults()
-                    .setSerializers(
-                            TypeSerializers.getDefaultSerializers().newChild().registerType(TypeToken.of(IpSet.class), new IpSet.IpSetSerializer()))
-                    .setHeader(HEADER));
-            this.configBase = this.configMapper.populate(this.root.getNode(this.modId));
+            // load settings from file
+            CommentedConfigurationNode loadedNode = this.loader.load();
+
+            // attempt to strip duplicate settings from the file
+            // (this only happens on the first pass of the file - see javadocs below)
+            if (cleanupConfig(loadedNode)) {
+                this.loader.save(loadedNode);
+            }
+
+            // store "what's in the file" separately in memory
+            this.fileData = loadedNode;
+
+            // make a copy of the file data
+            this.data = this.fileData.copy();
+
+            // merge with settings from parent
+            if (this.parent != null) {
+                this.data.mergeValuesFrom(this.parent.data);
+            }
+
+            // populate the config object
+            populateInstance();
         } catch (Exception e) {
             SpongeImpl.getLogger().error("Failed to load configuration", e);
         }
     }
 
+    private void populateInstance() throws ObjectMappingException {
+        this.configMapper.populate(this.data.getNode(this.modId));
```

Comment:
This needs to set mappedInstance or getConfig will cause NPE.

Reply:
Fixed